content/projects/python-specific/file-io/_index.md

```diff
@@ -3,12 +3,90 @@ _db_id: 695
 content_type: project
 flavours:
 - python
-ready: false
+ready: true
 submission_type: repo
 tags:
 - FileIO
 title: interacting with files
 ---
 
-TODO: take a look at this: content/projects/nodejs/file-io/_index.md 
-Make a projet spec that works
\ No newline at end of file
+You are required to create a back-end service that will help capture basic information about prospective students who come to inquire here at Umuzi. In this project you'll just be storing and retrieving information from plain old json files.
+
+## Instructions
+
+1. Create a class called Visitor. Instances of this class should have the following properties:
+
+- full name
+- age
+- date of visit
+- time of visit
+- comments
+- name of the person who assisted the visitor
+
+2. Create a function called `save` that saves the visitor's data to a JSON file. The file name should be named like this `visitor_{their_full_name}.json`.
+
+```
+alice.save()   # results in visitor_alice_cooper.json
+bob.save()     # results in visitor_bob_marley.json
+charlie.save() # results in visitor_charley_sheen.json
+```
+
+Notice that the full name used in the file is all lower-case and spaces are replaced by underscores.
+
+3. Create a function called `load` that takes in a name and then grabs a Visitor object from file. It should simply `print` the visitor.
```

Comment:
The reason we went with print for the js version of this is because node is weird about reading files - python is less weird. It would be better to have the load function just return a visitor instance. 

`Visitor.load(name)`

Reply:
Shouldn't this then be a function that works like this (assuming we have `visitor_alice_cooper.json`)?
```py
load("Alice Cooper") # gets info from visitor_alice_cooper.json and returns it
```

Reply:
It would make most sense as a `classmethod`